src/result_processor.c

```diff
@@ -86,6 +88,44 @@ typedef struct {
   size_t timeoutLimiter;    // counter to limit number of calls to TimedOut_WithCounter()
 } RPQueryIterator;
 
+
+/****
+ * Get_DocumentMetadata - get the document metadata for the current document from the iterator.
+ * If the document is deleted or expired, return false.
+ * If the document is not deleted or expired, return true.
+ * If the document is not deleted or expired, and dmd is not NULL, set *dmd to the document metadata.
+ * @param spec The index spec
+ * @param docs The document table
+ * @param sctx The search context
+ * @param it The query iterator
+ * @param dmd The document metadata pointer to set
+ * @return true if the document is not deleted or expired, false otherwise.
+ */
+static bool Get_DocumentMetadata(IndexSpec* spec, DocTable* docs, RedisSearchCtx *sctx, const QueryIterator *it, const RSDocumentMetadata **dmd) {
```

Comment:
The function name is a bit odd.

```suggestion
static bool getDocumentMetadata(IndexSpec* spec, DocTable* docs, RedisSearchCtx *sctx, const QueryIterator *it, const RSDocumentMetadata **dmd) {
```